opm/simulators/wells/WellInterface_impl.hpp

```diff
@@ -1137,6 +1137,10 @@ namespace Opm
         const auto& summaryState = simulator.vanguard().summaryState();
         const auto& schedule = simulator.vanguard().schedule();
 
+        // Discard old primary variables, the new well state
+        // may not be anywhere near the old one.
+        ws.primaryvar.resize(0);
```

Comment:
should we just call updatePrimaryVariable at the end of the this function instead so make sure the updated well state will be reflected in the primary variables?

Reply:
The problem is that if we do that, the primary variables will read the values stored in ws.primaryvar (which are not modified in this function) instead of setting the primary variables based on surface_rates etc. An alternative would be a boolean flag similar to what was in earlier versions of this, but the current version requires no extra data member so I prefer it.